File: backend-java/src/main/java/com/skewwhiffy/auraltester/dao/AbsoluteNoteDao.java
package com.skewwhiffy.auraltester.dao;

import com.skewwhiffy.auraltester.notation.model.note.AbsoluteNote;

import java.util.Optional;

public record AbsoluteNoteDao(
        NoteDao note,
        OctaveDao octave,
        String lyric
) {
    public AbsoluteNote toModel() {
        return new AbsoluteNote(note.toModel(), octave.toModel(), Optional.ofNullable(lyric));
    }
}

File: backend/src/main/kotlin/com/skewwhiffy/auraltester/dao/AbsoluteNoteDao.kt
package com.skewwhiffy.auraltester.dao

import com.skewwhiffy.auraltester.notation.model.note.AbsoluteNote

data class AbsoluteNoteDao(
    val note: NoteDao,
    val octave: OctaveDao,
    val lyric: String?
)

val AbsoluteNoteDao.model
    get() = AbsoluteNote(note.model, octave.model, lyric)
